### PR TITLE
fix(gui-app,protocol): offer the terminal sign-in on terminalLogin alone, not oauthArgs

### PR DESCRIPTION
**Pull request by hdkshingala**
Client half of traycerai/traycer-internal#5475 (host: launch the CLI in the sign-in terminal for Qwen, Droid, OMP and OpenCode). Follows #1699 / #1708.

## Problem

`providerSupportsTerminalLogin` required a non-empty `oauthArgs` beside `terminalLogin`, from when Copilot and Reasonix were the only terminal-login providers and both happened to carry one. The host now declares `terminalLogin` for four providers that have **no headless login command at all** (Qwen's `qwen auth` is removed and sign-in is `/auth` inside the TUI; Droid has only `droid doctor`; OMP's sign-in is `login <provider>` inside the TUI; OpenCode's `auth login` is an interactive picker). The host launches the CLI itself in the sign-in terminal, and their `oauthArgs` stays `null` because that is what keeps the **headless** button hidden on clients that predate `terminalLogin`. With the old predicate, that same `null` hid the **terminal** button for exactly those four.

## What changes

| File | Change |
| --- | --- |
| `provider-signin-availability.ts` | `providerSupportsTerminalLogin` reads `terminalLogin` alone. `providerSignInUnavailableHint` answers the terminal case **first**, before its "does not support browser sign-in, authenticate with its own CLI" branch, so a terminal-login provider with no headless command is never sent to a CLI Traycer can open for it. |
| `harness-model-picker-create-profile-gate.ts` | Already ordered terminal-first; comment no longer claims terminal-login providers "have real `oauthArgs`". |
| `provider-reauth-banner.tsx` | `deriveLoginOptions` unchanged in logic (terminal excludes headless; headless still needs a real subcommand); comments corrected. |
| `provider-setup-guidance.ts` | The generic terminal copy says the terminal "prints a sign-in code". For the launch-the-CLI providers what opens is the CLI's own UI, so `TERMINAL_SIGN_IN_COPY` re-words the summary, first step and hint to name the step inside it: Qwen "Type /auth …", Droid "Follow the sign-in prompt …", OMP "Type login followed by the provider …", OpenCode "Pick the provider and sign-in method …". Everything else stays generic (no manual command), so an old host that declares no capability still shows nothing for them, exactly like Copilot. |
| `protocol/src/host/provider-schemas.ts` | Doc-only: `terminalLogin`'s comment no longer calls `oauthArgs` the terminal's command source, and says a GUI gate reads this field alone. No schema change. |

Class sweep: every GUI reader of `oauthArgs` next to `terminalLogin` (the predicate, the hint, the profile gate, the banner) plus the protocol comment; `rg` for "real \`oauthArgs\`" / "command the terminal runs" finds no stale claim.

## Compatibility

- **New GUI, old host** (no `terminalLogin` for these four): unchanged, the predicate is false and nothing is offered.
- **New GUI, new host**: the picker CTA, the composer banner row and the Settings hint all offer / point at the terminal sign-in; the headless button stays hidden.
- Copilot and Reasonix are unaffected: they still carry `oauthArgs` and `terminalLogin`.

## Tests

Three cases that pinned the old coupling are flipped (predicate, profile gate, banner) and the null-`oauthArgs` terminal case is added for the predicate, the hint, the profile gate, the banner, and the guidance copy (per-provider steps, generic labels kept, null on a host with no capability). The two guidance tests that spelled "capability absent" as `oauthArgs: []` now spell it as `terminalLogin: null`, which is what absence is.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: clients/gui-app/src/components/home/pickers/__tests__/harness-model-picker-create-profile-gate.test.ts [hunk 1]
```diff
@@ -55,19 +55,20 @@ describe("resolveCreateProfileGate", () => {
...
 
-  // Unified gate: terminalLogin present but no real oauthArgs must read the
-  // same as "not a terminal-login provider" here too - the generic reason,
-  // not the terminal-specific one.
-  it("uses the generic reason, not the terminal one, when oauthArgs is empty", () => {
+  // A launch-the-CLI provider (Qwen, Droid, OMP, OpenCode) declares
+  // `terminalLogin` with `oauthArgs: null`. The terminal reason must still
+  // win: the generic one names browser sign-in, which these providers do not
+  // do either.
+  it("uses the terminal reason for a terminal-login provider with no oauthArgs", () => {
```
[the remark this answers is not in the document]
Comment (hdkshingala) [reply clients/gui-app/src/components/home/pickers/__tests__/harness-model-picker-create-profile-gate.test.ts]: Valid, and the boundary was worse off than the note says: this suite is not the only place it went missing. Swept the class across every consumer that asserts terminal-first behaviour, each now parametrized over `null` and `[]`:

| Suite | Case |
| --- | --- |
| `harness-model-picker-create-profile-gate.test.ts` | terminal reason, not the browser-sign-in one |
| `provider-signin-availability.test.ts` | hint points at the terminal flow, not "its own CLI" |
| `provider-reauth-banner.test.tsx` | terminal button offered, headless one not |
| `provider-setup-guidance.test.ts` | `resolveProviderTerminalSetup` returns the in-CLI copy |

The banner is why this was worth more than one case: its previous `[]` test asserted the OPPOSITE (neither button), so converting it to `null` dropped that boundary from the suite altogether — exactly the regression you describe, already half-materialised.

Invariant checked while doing it: all four consumers are now indifferent to `oauthArgs` for a terminal-login provider — `providerSupportsTerminalLogin` reads `terminalLogin` alone, and the two consumers that still branch on `oauthArgs` (`providerSignInUnavailableHint`, `resolveCreateProfileGate`) answer the terminal case first. 78 tests green in those four suites.

Fixed in 0831f2c4d.

File: clients/gui-app/src/lib/providers/provider-setup-guidance.ts [hunk 2]
```diff
@@ -110,27 +120,80 @@ export function providerSetupGuidance(
...
- * describe one flow the same way.
+ * describe one flow the same way. The launch-the-CLI providers re-word the
+ * three sentences that would otherwise describe a sign-in code nothing prints
+ * (`TERMINAL_SIGN_IN_COPY`) and keep everything else.
  */
 export function defaultTerminalSignInGuidance(
   providerId: ProviderId,
 ): ProviderSetupGuidance {
   const name = PROVIDER_DISPLAY_NAMES[providerId];
+  const copy = TERMINAL_SIGN_IN_COPY[providerId] ?? null;
```
[the remark this answers is not in the document]
Comment (hdkshingala) [reply clients/gui-app/src/lib/providers/provider-setup-guidance.ts:180]: Real, and thanks — the banner was reading the OVERRIDE table (`providerSetupGuidance`, Reasonix-only) and then falling back to its **own inline copy** of the generic two sentences, so `TERMINAL_SIGN_IN_COPY` could only ever reach the picker.

Fixed at the source rather than by teaching the banner about the new table, because the duplicate literals were the actual defect:

```ts
export function providerTerminalGuidance(providerId: ProviderId): ProviderSetupGuidance {
  return providerSetupGuidance(providerId) ?? defaultTerminalSignInGuidance(providerId);
}
```

`resolveProviderTerminalSetup` (picker CTA, auth line, empty state) and `TerminalLoginRow` (banner) now both resolve through it, so the sentences exist once. The banner's inline fallback is gone, and with it `TerminalLoginRow`'s `providerLabel` prop — that fallback was its only user.

Invariant pass: the early-return branch in `resolveProviderTerminalSetup` still reads `providerSetupGuidance` directly, because there it must return `null` for a provider with no override on a host that declares no capability — the new resolver would have handed back generic copy for a button that does not exist. No surface renders it for a provider without `terminalLogin`: the capability gate decides that, and both call sites are behind it.

Also corrected the module doc, which claimed the generic copy was already "the same sentences the composer banner uses" — true when written, false the moment a provider needed different ones. It now names the resolver that makes it true.

Regression test: the banner renders "Qwen Code opens in that terminal. Type /auth …" and no "sign-in code" text. 89 tests green across the banner, guidance and two picker suites.

Fixed in 0f0835493.
